packages/api/lib/distribution.js

```diff
@@ -144,32 +196,45 @@ async function ensureAuthorizedOrRedirect(req, res, next) {
   const redirectURLForAuthorizationCode = oauthClient.getAuthorizationUrl(req.path);
   const accessToken = req.cookies.accessToken;
 
-  if (!accessToken) return res.redirect(307, redirectURLForAuthorizationCode);
-
+  let authorizedMetadata;
   let accessTokenRecord;
-  try {
-    accessTokenRecord = await accessTokenModel.get({ accessToken });
-  } catch (error) {
-    if (error instanceof RecordDoesNotExist) {
-      return res.redirect(307, redirectURLForAuthorizationCode);
+  if (accessToken) {
+    try {
+      accessTokenRecord = await accessTokenModel.get({ accessToken });
+      authorizedMetadata = {
+        userName: accessTokenRecord.username,
+        userGroups: get(accessTokenRecord, 'tokenInfo.user_groups', []),
+      };
+    } catch (error) {
+      if (!(error instanceof RecordDoesNotExist)) {
```

Comment:
And what happens if it is a Record does not exist error?

Reply:
accessTokenRecord is undefined.  The function continues to check other conditions.